### ISSUE TITLE
Support v3 dependencies for pwsh packages

### ISSUE THREAD

**Issue by nbarnwell** [PowerShell/PSResourceGet#1575]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/Microsoft.PowerShell.PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

1. Register an authenticated Azure DevOps Artifacts `PSResourceRepository` that has the PowerShell Gallery configured as an upstream source. Use the `Microsoft.PowerShell.SecureVault` module as the source of credentials.
2. Unregister the PowerShell Gallery resource repository.
3. `Install-PSResource PSWindowsUpdate`

### Expected behavior

```console
The `PSWindowsUpdate` module and any dependencies are installed.
```

### Actual behavior

```console
The install appears to work, but the following warning is shown:

> WARNING: Installing dependencies is not currently supported for V3 server protocol repositories. The package will be installed without installing dependencies.
```

### Environment data

```powershell
PS C:\temp> Get-Module Microsoft.PowerShell.PSResourceGet; $PSVersionTable

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     1.0.2                 Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PSResourceRepository, Get-PSScriptFileInfo…}

Key   : PSVersion
Value : 7.4.0
Name  : PSVersion


Key   : PSEdition
Value : Core
Name  : PSEdition


Key   : GitCommitId
Value : 7.4.0
Name  : GitCommitId


Key   : OS
Value : Microsoft Windows 10.0.19044
Name  : OS


Key   : Platform
Value : Win32NT
Name  : Platform


Key   : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0…}
Name  : PSCompatibleVersions


Key   : PSRemotingProtocolVersion
Value : 2.3
Name  : PSRemotingProtocolVersion


Key   : SerializationVersion
Value : 1.1.0.1
Name  : SerializationVersion


Key   : WSManStackVersion
Value : 3.0
Name  : WSManStackVersion
```

**Comment by SydneyhSmith** [comment]
Thanks @nbarnwell  what repository are you seeing this with? and what is the apiversion?

**Comment by nbarnwell** [comment]
I updated the reproduction steps.

**Comment by SydneyhSmith** [comment]
Thanks, if you register this as a v2 feed instead of a v3 feed you can get rid of this warning message

**Comment by nbarnwell** [comment]
I'll try it again but I'm pretty sure I couldn't get it to work at all with a v2 feed, I think because it's an authenticated feed.

**Comment by nbarnwell** [comment]
@SydneyhSmith Indeed it doesn't work for authenticated repositories. In fact, I get the error I previously reported in https://github.com/PowerShell/PSResourceGet/issues/1491, which I don't think anyone cares about, either. PowerShell module stuff isn't fit for purpose when it comes to authenticated feeds. It's all very well and good encouraging folks to publish to the PowerShell gallery, but to have this massive blind spot when it comes to corporate packages in private feeds is a real surprise. I'd have thought the PowerShellGet team would've been dogfooding on Azure DevOps Artifacts feeds themselves, but I guess not or these issues (there are several) would get looked at.

Also, I wasn't trying to get rid of a warning message - I need dependency support on v3 feeds. V2 feeds are broken so telling me to go back to that when I've been on a several-month-long journey trying to get all this to hang together is a bit hard to swallow.

**Comment by Gijsreyn** [comment]
I don't know if this issue is related to the topic here. 

What I find strange, is when adding PowerShellGallery as upstream source, it works with the old PowerShellGet to install a module through there. When using Install-PSResource, if the module has not been downloaded from PowerShellGallery, it throws an error it cannot find the module.

**Comment by SydneyhSmith** [comment]
@Gijsreyn can you please open a new issue for that-- that is not expected

**Comment by Gijsreyn** [comment]
@SydneyhSmith I'll run some more test to see the different endpoint urls from Azure Artifacts and open a issue if it persists.

**Comment by FriedrichWeinmann** [comment]
Heya, also blocked on this for a project :(
One note: I have this exact same issue, whether I have an upstream configured or not.

**Comment by nbarnwell** [comment]
Is there a better github issue to watch for progress on implementing this feature? I'm conscious that lots of people must have the same problem and this could have been a duplicate all along?

**Comment by SydneyhSmith** [comment]
on our wishlist -- hoping to get to this one

**Comment by codekaizen** [comment]
As of [PR 1023](https://github.com/PowerShell/PSResourceGet/pull/1023), PSResourceGet moved off the Nuget API and this appears to have degraded v3 nuget feed functionality substantially. Looking through the PSResourceGet issues, there are a number of them that relate to missing functionality in v3 feeds like installing dependencies, searching by tags (or almost anything besides ID), and authentication. @SydneyhSmith could you help us get visibility into what is the roadmap for PSResourceGet nuget v3 feed support, or really PSResourceGet in general? Things here and at https://devblogs.microsoft.com/powershell/ have gotten very quiet over the past year.

**Comment by MJECloud** [comment]
@SydneyhSmith Any update on this? Would be great to have this, would dependent module installation from Azure Artifacts f.e. a lot easier. 
I saw that there is already a pull request for this, any chance to look at this? 

Thanks in advance!

**Comment by et1975** [comment]
This rewrite of the package management has been a long-time coming disappointment. So many other basic things broken, but this? Not a word in the docs about the lack of dependencies support. Can't believe 1es is recommending this instead of getting you to fix v2 package manager to address s360 items.

**Comment by powercode** [comment]
Do you happen to have any updates here? This is fundamental.

**Comment by MJECloud** [comment]
@powercode this should be available in the current preview.

**Comment by nbarnwell** [comment]
Does that mean that if I `Install-module PSResourceGet -RequiredVersion X` this will start working? What version do I need to specify?
